### ISSUE TITLE
cmd/pprof: update the vendored tool from upstream

### ISSUE THREAD

**Issue by ALTree** [golang/go#21047]
Opening this to document a recent (failed) effort to update `pprof` from upstream.

The tool was successfully updated from upstream on March 1 ([CL 37652](https://golang.org/cl/37652)). A failed attempt to a final update for go1.9 was made on June 20 ([CL 46155](https://golang.org/cl/46155)). The update broke several builders and was immediately reverted.

The failing builders:

- We had an [OSX10.8 failure](https://build.golang.org/log/e9207dc8ed9dd01f11938974217b26bf2b901715), an [OSX10.10 failure](https://build.golang.org/log/bfb7c4db36ddd4a4f801d5b33cfe8088b1a80c73), and failures on the [`darwin-arm-a1549ios`](https://build.golang.org/log/e4a96f0bbced472564e20e3258a7458d4a0a0859) and [`darwin-arm64-a1549ios`](https://build.golang.org/log/94be81b79f30c7d74ef7829fcb205a4197f59db7) builders. All these failures were on the recently introduced **`TestHttpsInsecure` test**. I reported the issue upstream (google/pprof#146).

- Both the android builders were failing ([failure1](https://build.golang.org/log/153a895afec205aee960dc65c867a47e2d577ad0), [failure2](https://build.golang.org/log/e391e89a9026831f9481dda049ad0fff47dfcf8b)), again, on `TestHttpsInsecure`, but with a different error (related to filesystem permissions, apparently).

- The `linux-arm-arm5spacemonkey` was [failing](https://build.golang.org/log/05399ae28ec1c374085b71b896200c4c06af1578) on the `TestHttpsInsecure` test, with the same error we got on the android builders (issues with filesystem permissions).

 - Two of the plan9 builders (`plan9-386` and `plan9-arm`) were failing ([failure1](https://build.golang.org/log/e6b43820b8e957cda2c7d2692ca9590eeb259b79), [failure2](https://build.golang.org/log/e03d632fdc4a1751623dacaa7dd3754af98348e9)), again, on `TestHttpsInsecure`, with the same error we got on the darwin builders.

To sum it up: all the failures are on the `TestHttpsInsecure` test. Darwin and plan9 builders are failing because the test somehow generates bad/unexpected profiles(?). A `linux-arm5` builder and both the android builders are failing with errors related to bad filesystem permission (I suspect the test makes assumptions about write permissions on certain folders that do not hold on some of the builders).

**Comment by aalexand** [comment]
https://github.com/google/pprof/pull/169 should fix the file system related issue on Android.

`TestHttpsInsecure` test is special in that it is the only test that collects the data by starting a Go server and a spinning goroutine and profiling the server. Hence these new issues found.

Is it possible to add building and testing of google/pprof at tip to the Go build matrix, "Sub-repositories at tip" section? This would continuously ensure that pprof is Go vendoring ready.

**Comment by ALTree** [comment]
> Is it possible to add building and testing of google/pprof at tip to the Go build matrix, "Sub-repositories at tip" section? This would continuously ensure that pprof is Go vendoring ready.

@bradfitz is it possible/do we want this?

**Comment by bradfitz** [comment]
We want that, yes. And everything's possible with a bit of time. We're short on time. But if somebody wants to work on #14594, that'd be great.

**Comment by aalexand** [comment]
@ALTree google/pprof#146 was fixed, can you try updating pprof in Go vendoring area again?

**Comment by ALTree** [comment]
@aalexand nice! I could send a change but it won't be merged now because we're deep deep into the code freeze. I'll send a change as soon as the tree re-open for go1.10 in August.

**Comment by aalexand** [comment]
@ALTree Could you create the change just to see if all the tests pass?

**Comment by ALTree** [comment]
I can do that, but what platforms are you interested in? For example, we have an OSX10.8 builder but not an OSX10.8 trybot. Builders run on master after a commit, trybots can be used to test a change before committing it. 

So even if I upload a change and run the trybots on it, they won't tell us if the OSX10.8 issue is gone. In fact, with the old failed update in June the trybots said "OK" on my change and only after it was merged I noticed the OSX10.8, OSX10.10 and iPhone failures (by looking at the builders webpage).

**Comment by aalexand** [comment]
I see. Then it doesn't make much sense, no. I didn't know the set of trybot envs is limited.

**Comment by ALTree** [comment]
pprof was vendored and the builders are happy (except the plan9 ones), so I'm closing this issue.

I opened  #22559 for the plan9 issue.
